src/main/java/org/opensearch/security/user/InternalUserTokenHandler.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.user;
+
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.identity.tokens.AuthToken;
+import org.opensearch.identity.tokens.BasicAuthToken;
+import org.opensearch.identity.tokens.TokenManager;
+import org.opensearch.security.securityconf.Hashed;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+
+import static org.opensearch.security.dlic.rest.support.Utils.universalHash;
+
+public class InternalUserTokenHandler implements TokenManager {
```

Comment:
This class looks like its doing too much. In the PR description it mentions that this PR is for `issueOnBehalfOfToken`. Are all of these methods being utilized in this class?